src/Features/Core/Portable/Completion/CompletionService_GetCompletions.cs

```diff
@@ -268,12 +270,23 @@ private static bool HasAnyItems(CompletionContext cc)
                 return CompletionList.Empty;
             }
 
-            return CompletionList.Create(
-                completionContexts[0].CompletionListSpan,   // All contexts have the same completion list span.
-                displayNameToItemsMap.SortToSegmentedList(),
-                GetRules(options),
-                suggestionModeItem,
-                isExclusive);
+            var list = s_sortListPool.Allocate();
+            try
+            {
+                displayNameToItemsMap.PopulateSortedList(list);
+
+                return CompletionList.Create(
+                    completionContexts[0].CompletionListSpan,   // All contexts have the same completion list span.
+                    list,
```

Comment:
We can't pass in a pooled list to create CompletionList here.

The pool we had before this change already uses List, which is where the sorting happened. We only add sorted items to SegmentedList (to avoid LOH allocation) so the slowness of sorting a SegmentedArray shouldn't be an issue here.

On the other hand, I think we might be able to avoid sorting items in returned list completely, since itemMananger will be sorting them later anyway. This can be done by adding an CompletionOption

Reply:
Whoa, I swear I followed into the Create method and that it didn't store the value, but you're right that it does. Let me look back into this.

Reply:
We tried to accommodate `SegmentList` ([via IReadonlyList](https://github.com/dotnet/roslyn/issues/61971)) in `CompletionList` to avoid LOH allocation caused by enabling import completion. The fact that we are trying to get though two layers (Roslyn CompletionService vs Editor AsyncCompletionSource) to get completion and both are public APIs definitely made things more complicated.